backend/server.py

```
@@ -1086,8 +1086,204 @@ async def admin_update_affiliate(
+    if not payload.terms_accepted:
+        raise HTTPException(status_code=400, detail="Terms acceptance required")
+    if not payload.newsletter_subscribed:
+        raise HTTPException(status_code=400, detail="Newsletter subscription required")
+
+    clean = sanitize_signup_payload(payload)
+
+    result = await session.execute(select(NewsletterSubscriber).where(NewsletterSubscriber.email == clean["email"]))
+    if result.scalar_one_or_none():
+        raise HTTPException(status_code=400, detail="Email already registered")
+
+    plaintext_pin = clean["pin"] or f"{random.randint(0, 999999):06d}"
```

Ruff (0.14.14)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)